src/main/kotlin/network/cere/telegram/bot/CampaignChatCacheService.kt

```diff
@@ -0,0 +1,159 @@
+                val campaignsResponse = robClient.getCampaigns(
+                    dataServiceId = config.appId(),
+                    organizationId = organization.id
+                )
+                allCampaigns.addAll(campaignsResponse.data)
+            }
+
+            val newMap = allCampaigns.flatMap { campaign ->
+                val formData = campaign.formData
+                val chatConfigs: List<ChatChallengeConfig> = extractChatIdsFromFormData(formData)
+                chatConfigs.map { chatConfig ->
+                    chatConfig.chatId to CampaignContext(
+                        orgId = campaign.appId?.toLongOrNull() ?: 0L,
+                        campaignId = campaign.campaignId.toLong(),
+                        campaignName = campaign.campaignName ?: "Unknown Campaign",
+                        challengeSettings = chatConfig.challengeSettings
+                    )
+                }
+            }.toMap()
+
+            cache = newMap
+            lastUpdated = Instant.now()
+        } catch (e: Exception) {
+            log.error("❌ Failed to refresh campaign-chat mapping", e)
+        }
+    }
+
+    private fun extractChatIdsFromFormData(formData: String?): List<ChatChallengeConfig> {
+        if (formData == null) return emptyList()
+
+        return try {
+            val json = Json { ignoreUnknownKeys = true }
+            val root = json.parseToJsonElement(formData).jsonObject
+
+            val chatField = root["telegramChannels"] ?: return emptyList()
+            val challengeSettingsObj = root["challengeSettings"]?.jsonObject
+
+            val cooldown = challengeSettingsObj?.get("cooldownHours")?.jsonPrimitive?.intOrNull ?: 24
+            val maxImageGeneration = challengeSettingsObj?.get("maxImageGenerationPerDay")?.jsonPrimitive?.intOrNull ?: 1
+            val maxBoost = challengeSettingsObj?.get("maxBoostPerDay")?.jsonPrimitive?.intOrNull ?: 1
+            val maxChannelRequestsPerDay = challengeSettingsObj?.get("maxChannelRequestsPerDay")?.jsonPrimitive?.intOrNull ?: 1000
+
+            // Extract prompt tags
+            val promptTagsArray = challengeSettingsObj?.get("promptTags")?.jsonArray
+            val promptTags = if (promptTagsArray != null) {
+                promptTagsArray.mapNotNull { element ->
+                    if (element is JsonObject) {
+                        val tag = element["tag"]?.jsonPrimitive?.content
+                        val prompt = element["prompt"]?.jsonPrimitive?.content
+                        if (tag != null && prompt != null) {
+                            PromptTag(tag, prompt)
+                        } else null
+                    } else null
+                }
+            } else {
+                listOf(
+                    PromptTag("fire", "anime-style portrait of a man in a tuxedo with a blazing fire aura behind him, surrounded by heat waves and glowing embers, dramatic lighting, fiery background, highly detailed, cinematic look"),
```

Comment:
at least move to constants. better to config file. or remove at all and make them 'required' on campaign configurator side